src/app/shared/components/ride-info/ride-info.component.ts

```diff
@@ -0,0 +1,98 @@
+import { HttpClient } from '@angular/common/http';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, effect, inject } from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
+import { ModalRideInfo, Segment } from '@shared/interfaces/route-info.interface';
+import { calculateStopDuration } from '@shared/utils/calculate-train-stop-duration';
+import { formatTime } from '@shared/utils/format-date';
+import { TuiDialogContext } from '@taiga-ui/core';
+import { POLYMORPHEUS_CONTEXT } from '@taiga-ui/polymorpheus';
+// TODO: import Station and etc. interfaces if availiable
```

Comment:
You can use Station interface in feature admin instead of your Station inteface, instead of ConnectedStation -  interface ConnectedStation in admin feature, instead of StationInfo - interface Trip in feature search